int32 theta_min
int32 theta_max
int32 alpha_min
int32 alpha_max
